assignment1/deployments/deploymentCrt.go

```diff
@@ -0,0 +1,110 @@
+package deployments
+
+import (
+	"context"
+	"fmt"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+	"log"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var rtc client.Client
+var deploy = &appsv1.Deployment{}
+
+func SetRtcClient(rtClient client.Client) {
+	rtc = rtClient
+}
+
+func CreateRtcDeployment() {
+	
+	deploy := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "rtc-deployment",
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: pointer.Int32Ptr(1),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": "nginx",
+				},
+			},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app": "nginx",
+					},
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Name:  "rtc-deployment-container",
+							Image: "nginx",
+							Ports: []v1.ContainerPort{
+								{
+									Name:          "port1",
+									Protocol:      v1.ProtocolTCP,
+									ContainerPort: 80,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	fmt.Println("Creating Deployment...")
+	err := rtc.Create(context.Background(), deploy)
```

Comment:
Should return error in case of error. Modify the function signature to return an error